File: Labs/Lab 6/CS/Lab6/TextContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class TextContainer
{
    private List<StringClass> strings;

    public TextContainer()
    {
        strings = new List<StringClass>();
    }

    public void AddString(StringClass str)
    {
        strings.Add(str);
    }

    public void RemoveString(StringClass str)
    {
        strings.Remove(str);
    }

    public void ClearText()
    {
        strings.Clear();
    }

    public int CountCharacters()
    {
        return strings.Sum(str => str.Content.Length);
    }

    public int FindString(string searchStr)
    {
        return strings.Count(str => str.Content.Contains(searchStr));
    }

    public void ReplaceCharacterInText(char oldChar, char newChar)
    {
        foreach (var str in strings)
        {
            str.ReplaceCharacter(oldChar, newChar);
        }
    }
}

File: Labs/Lab 6/TS/TextContainer.ts
import { StringClass } from "./StringClass";

export class TextContainer {
    private strings: StringClass[];

    constructor() {
        this.strings = [];
    }

    addString(str: StringClass): void {
        this.strings.push(str);
    }

    removeString(str: StringClass): void {
        this.strings = this.strings.filter(s => s !== str);
    }

    clearText(): void {
        this.strings = [];
    }

    countCharacters(): number {
        return this.strings.reduce((acc, str) => acc + str.getContent().length, 0);
    }

    findString(searchStr: string): number {
        return this.strings.filter(str => str.getContent().includes(searchStr)).length;
    }

    replaceCharacterInText(oldChar: string, newChar: string): void {
        this.strings.forEach(str => str.replaceCharacter(oldChar, newChar));
    }
}
